src/domains/wallet-app/api/payment-wallet/v1/_post_wallets_with_pm_policy.xml.tpl

```diff
@@ -1,6 +1,68 @@
 <policies>
     <inbound>
+      <!-- Extract payment method name for create redirectUrl -->
       <set-variable  name="walletToken"  value="@(context.Request.Headers.GetValueOrDefault("Authorization", "").Replace("Bearer ",""))"  />
+      <set-variable name="requestBody" value="@(context.Request.Body.As<JObject>(preserveContent: true))" />
+      <set-variable name="paymentMethodId" value="@((string)((JObject) context.Variables["requestBody"])["paymentMethodId"])" />
+      <send-request ignore-error="false" timeout="10" response-variable-name="paymentMethodsResponse">
+          <set-url>@("https://${ecommerce-basepath}/pagopa-ecommerce-payment-methods-service/payment-methods/" + context.Variables["paymentMethodId"])</set-url>
+          <set-method>GET</set-method>
+          <set-header name="x-client-id" exists-action="override">
+              <value>IO</value>
+          </set-header>
+      </send-request>
+      <choose>
+          <when condition="@(((IResponse)context.Variables["paymentMethodsResponse"]).StatusCode != 200)">
+              <return-response>
+              <set-status code="502" reason="Bad Gateway" />
+              <set-header name="Content-Type" exists-action="override">
+                  <value>application/json</value>
+              </set-header>
+              <set-body>
+                  {
+                      "title": "Error retrieving eCommerce payment methods",
+                      "status": 502,
+                      "detail": "There was an error retrieving eCommerce payment methods"
+                  }
+              </set-body>
+              </return-response>
+          </when>
+      </choose>
+      <set-variable name="paymentMethodsResponseBody" value="@(((IResponse)context.Variables["paymentMethodsResponse"]).Body.As<JObject>())" />
+      <set-variable name="paymentMethodTypeCode" value="@((string)((JObject) context.Variables["paymentMethodsResponseBody"])["paymentTypeCode"])" />
+      <set-variable name="redirectUrlPrefix" value="@{
+              string returnedPaymentMethodTypeCode = (string)context.Variables["paymentMethodTypeCode"];
+              var paymentMethodTypeCodes = new Dictionary<string, string>
+                  {
+                      { "CP", "pm-onboarding/creditcard" },
+                      { "BPAY", "pm-onboarding/bpay" },
+                      { "PPAY", "pm-onboarding/paypal" }
+                  };
+
+              if (paymentMethodTypeCodes.ContainsKey(returnedPaymentMethodTypeCode)) {
+                  return paymentMethodTypeCodes[returnedPaymentMethodTypeCode];
+              }
+              return "";
```

Comment:
```suggestion
              string redirectUrlPrefix;
              paymentMethodTypeCodes.TryGetValue(returnedPaymentMethodTypeCode, out redirectUrlPrefix);
              return redirectUrlPrefix;
```
This piece of code can be simplified as follow

Reply:
Fixed with 426c6c37aa6492574cc6c913032c8156aefb752b